### PR TITLE
Jetpack checklist: put more focus on checklist

### PR DESCRIPTION
**Pull request by simison**
Put more focus on "thank you" and the checklist when landing on Calypso-proper first time during Jetpack onboarding.

#### Changes proposed in this Pull Request
While "thank you" -component is visible:
* Hide "feature list"
* Make checklist less visible
* Hide sidebar & section navigation

image

Resolves https://github.com/Automattic/wp-calypso/issues/32171

#### TODO / can be done in follow up PRs
- Possibly add a `noGradient` to `FeatureExample` component or add a new component to hide content like in designs instead of using gradient fadeout (pending https://github.com/Automattic/wp-calypso/issues/32171#issuecomment-491333788)
- Improve mobile size

#### Testing instructions

Just to test how it looks like:
- Have a Jetpack site with a paid plan
- Open `http://calypso.localhost:3000/plans/my-plan/:YOUR_SITE?thank-you`

Test the flow
- Click through the flow, do you see thank you without all the surrounding UI but when clicking "hide message" or "continue", all that UI comes back?
- Test also free plan since it shouldn't direct you via "thank you" flow
- Test that non-Jetpack flows are unaffected
- Check that things look like expected on mobile, too.

### CODE REVIEW

**Comment by simison** [comment]
Mobile views need some thinking — could we hide the checklist completely on small screens?

image

image

**Comment by simison** [comment]
Thanks @sirreal — hiding the sidebar completely instead of showing it partially is a good idea I think, but I don't think that's what @jeffgolenski had in mind. I think he wanted to have some kind of "placeholder" visible and that's why it was in his designs.

I'd go with @sirreal 's suggestion because there's less visual noise and it feels more balanced. The customer never saw the sidebar up to that point yet anyway, so a white placeholder doesn't necessarily communicate clearly that _"just wait, there's a sidebar coming here later on"_. Now the step when the sidebar then finally does appear is a bit bigger visual "jump", though.

Comparison:

image

Screenshot 2019-05-14 at 14 17 29

I'd like to then hide also the section navigation placeholder.

Thoughts @jeffgolenski ?

File: client/my-sites/plans/current-plan/index.jsx [hunk 1]
```diff
@@ -155,18 +158,28 @@ class CurrentPlan extends Component {
...
-				<div
-					className={ classNames( 'current-plan__header-text current-plan__text', {
-						'is-placeholder': { isLoading },
-					} ) }
-				>
-					<h1 className="current-plan__header-heading">{ planFeaturesHeader }</h1>
-				</div>
-				<ProductPurchaseFeaturesList plan={ currentPlanSlug } isPlaceholder={ isLoading } />
+				{ ! showThankYou && (
+					<Fragment>
```
[the remark this answers is not in the document]
Comment (simison) [reply client/my-sites/plans/current-plan/index.jsx]: Let's do that once we can remove `showJetpackChecklist` flag and sorting out these conditionals becomes much more straightforward?

**Comment by jeffgolenski** [comment]
@simison 

Per your questions here and in slack:

1) I'd like to initially try having a sidebar placeholder to see how it feels, so there isn't so much page "jumping" after the focus leaves the checklist
2) same goes for section placeholder. I'd like to have both of those in place so there's no visual jump after the checklist focus goes away
3) Yes, let's do the gradient fade on the checklist!

**Comment by jeffgolenski** [comment]
Tested and works great. Thanks @simison

**Comment by shaunandrews** [comment]
This feels incredibly strange — almost like the page is still loading, or there's a rendering bug.

I would think a modal overlay would be way more appropriate than hiding bits of our UI.

image

Or a separate screen altogether, similar to what @sirreal suggested.

image

--

Perhaps there's a reason for having the checklist and filterbar visible, but obscured — but its not obvious.

**Comment by fditrapani** [comment]
> This feels incredibly strange — almost like the page is still loading, or there's a rendering bug.

I'll plus one this too. It seems like a new pattern we don't have anywhere else in Calypso. I like Shaun's idea as an alternative using components we already have.

**Comment by jeffgolenski** [comment]
After going through the flow a few more times, I also agree.

Sorry @simison - is it possible to whip up another PR where we remove the sidebar and section nav placeholder as you and everyone else agreed with before? My idea didn't work out! :-(

**Comment by simison** [comment]
No problem! :-) Both point-of-views were valid. Here we go: https://github.com/Automattic/wp-calypso/pull/33100

**Pull request merged**